packages/core/server/admin-api/services/query-layer-decorator.ts

```diff
@@ -28,8 +28,8 @@ const decorator = (service: IDecoratedService) => ({
     }
 
     await Promise.all(languages.map(async (lang) => {
-      const urlPattern = await getPluginService('urlPatternService').findByUid(uid, lang);
-      const languageRelations = getPluginService('urlPatternService').getRelationsFromPattern(urlPattern);
+      const urlPatterns = await getPluginService('urlPatternService').findByUid(uid, lang);
+      const languageRelations = urlPatterns.flatMap((urlPattern) => getPluginService('urlPatternService').getRelationsFromPattern(urlPattern));
```

Comment:
Why do we have to use a `flatMap` here while in the `bulk-generate` service we call the `getRelationsFromPattern` directly. That seems like a bug. We should probably just call it directly, the same way we do for the bulk generate service.

Reply:
This one still needs to be done.